pcdsdevices/targets.py

```diff
@@ -260,3 +267,795 @@ def move(self, nxspaces, nyspaces, wait=False):
+            First 4 -> alpha, last 4 -> beta
+        """
+        self._coefficients = coefficients
+
+    def get_samples(self, path=None):
+        """
+        Get all the available sample grids names that are currently saved.
+
+        Returns
+        -------
+        samples : list
+            List of strings of all the sample names available.
+        """
+        path = path or self._path
+        with open(path) as sample_file:
+            try:
+                data = yaml.safe_load(sample_file)
+                if not data:
+                    logger.info('The file is empty, no samples saved yet.')
+                    return
+                return list(data.keys())
+            except yaml.YAMLError as err:
+                logger.error('Error when loading the samples yaml file: %s',
+                             err)
+
+    def get_current_sample(self):
+        """
+        Get the current sample that is loaded.
+
+        Returns
+        -------
+        sample : dict
+            Dictionary with current sample information.
+        """
+        return self._current_sample
+
+    def set_current_sample(self, sample_name):
+        """
+        Set the current sample.
+
+        Parameters
+        ----------
+        sample_name : str
+            The name of the sample to be set as current one.
+        """
+        self._current_sample = str(sample_name)
+
+    def load_sample(self, sample_name, path=None):
+        """
+        Get the sample information and populate these parameters.
+
+        This function displays the parameters for the sample just loaded, but
+        also populates them, in the sense that it sets the current
+        `coefficients` and current `m, n` values.
+
+        Parameters
+        ----------
+        sample_name : str
+            Name of the sample to load.
+        path : str, optional
+            Path where the samples yaml file exists.
+        """
+        path = path or self._path
+        m_points, n_points, coeffs = self.get_sample_map_info(str(sample_name),
+                                                              path)
+        self.m_n_points = m_points, n_points
+        self.coefficients = coeffs
+        # make this sample the current one
+        self.set_current_sample(str(sample_name))
+        # TODO: might need to the set presets here tooo??
+
+    def get_sample_data(self, sample_name, path=None):
+        """
+        Get the information for a saved sample.
+
+        Parameters
+        ----------
+        sample_name : str
+            The sample name that we want the grid for. To see current
+            available samples call `mapped_grids`
+        path : str, optional
+            Path to the `.yml` file. Defaults to the path defined when
+            creating this object.
+
+        Returns
+        -------
+        data : dictionary
+            Dictionary of all the information for a saved sample.
```

Comment:
This can also return `None` if there are errors. You should do one of:
- Document that it can return `None`
- Make it raise errors
- Make it always return a `dict`

Reply:
Ok thanks, I think I covered most of (hopefully all but will double check) the ambiguous returns in 393ee51